lib/tasks/migrate.rake

```diff
@@ -261,6 +261,32 @@ namespace :migrate do
     end
   end
 
+  namespace :media do
+    # Media resources id 1 - 457799 are all subjects for project Snapshot Supernova.
+    # They currently exist outside of our media folder organizational structure.
+    # All files have already been copied over to /subject_location, this task must be run
+    # to update the media record src (which points to where the file is located)
+    desc 'Update src to subject_locations for Snapshot Supernova media'
+    task rewrite_supernova_file_locations: :environment do
+      # regex to check whether current src is an old path
+      OLD_PATH_REGEX = /\Apanoptes-uploads.zooniverse.org\/1\/0\//.freeze
+      # this will return match data that will give us just the file name,
+      # which is the part of the string following the old path
+      CAPTURE_GROUP_REGEX = /\A(?:panoptes-uploads.zooniverse.org\/1\/0\/)?(.+)\z/.freeze
+
+      puts 'starting supernova src location rewrite'
+      Media.where('id < 457800').find_each do |medium|
+        if OLD_PATH_REGEX.match(medium.src)
+          file_name = medium.src.match(CAPTURE_GROUP_REGEX)[1]
```

Comment:
```suggestion
        if matches = CAPTURE_GROUP_REGEX.match(medium.src)
          file_name = matches[1]
```
```
test = 'panoptes-uploads.zooniverse.org/1/0/file-name.jpeg'
 => "panoptes-uploads.zooniverse.org/1/0/file-name.jpeg" 
2.5.3 :030 > regex = /\A(?:panoptes-uploads.zooniverse.org\/1\/0\/)?(.+)\z/
 => /\A(?:panoptes-uploads.zooniverse.org\/1\/0\/)?(.+)\z/ 
2.5.3 :031 > if regex_capture_groups = regex.match(test)
2.5.3 :032?>     regex_capture_groups[1]
2.5.3 :033?>   else
2.5.3 :034?>     puts 'why you no match'
2.5.3 :035?>   end
 => "file-name.jpeg" 
```

Reply:
Original regex `/\A(?:panoptes-uploads.zooniverse.org\/1\/0\/)?(.+)\z/` would not have filtered out strings that did not start with `panoptes-uploads.zooniverse.org/1/0/` (due to that capture group starting with `?:`) but Ive updated it as needed so we are good.